opm/grid/polyhedralgrid/grid.hh

```diff
@@ -1013,10 +1017,19 @@ namespace Dune
             const int faceVertex = grid_.face_nodes[ grid_.face_nodepos[seed.index() ] + crner ];
             return copyToGlobalCoordinate( grid_.node_coordinates + GlobalCoordinate :: dimension * faceVertex );
           }
-        case dim:
+        case 2:
           {
-            const int coordIndex = GlobalCoordinate :: dimension * seed.index();
-            return copyToGlobalCoordinate( grid_.node_coordinates + coordIndex );
+            if (dim == 2) {
+              const int coordIndex = GlobalCoordinate :: dimension * seed.index();
+              return copyToGlobalCoordinate( grid_.node_coordinates + coordIndex );
+            }
+          }
+        case 3:
+          {
+            if (dim == 3) {
+              const int coordIndex = GlobalCoordinate :: dimension * seed.index();
+              return copyToGlobalCoordinate( grid_.node_coordinates + coordIndex );
+            }
```

Comment:
Same thing here. Code is less readable and has more code duplication. What is the reason?

I suggest to leave this as is.